src/VisualStudio/CSharp/Impl/CSharpVSResources.resx

```diff
@@ -637,4 +637,7 @@
   <data name="Allow_bank_line_after_colon_in_constructor_initializer" xml:space="preserve">
     <value>Allow blank line after colon in constructor initializer</value>
   </data>
-</root>
+  <data name="Move_to_end_of_expression_on_semi_colon" xml:space="preserve">
+    <value>Move to end of expression on semi-colon</value>
```

Comment:
So lets talk naming. I know internally this is called "Statement completion" but if I wasn't a Roslyn developer, and I wasn't for a long time, I would have no idea what that means if I saw it in the options screen, and I probably wouldn't want to touch the option just in case.

"Move to end of expression on semi-colon" is clear, but obviously suffers in that its not _quite_ everything this does, since I think it can also insert braces, right?

Anyway, bikeshed away.

Reply:
I think the name gets the idea across enough that I'd be able to figure out that turning it off would stop the behavior I didn't like. Although maybe combine the two and get "Complete statement on semicolon"? (My understanding is the long-term plan for the statement completion feature includes more than just the semicolon movement.)

Pretty nitpicky, but looking through the rest of this file and [VSPackage.resx](https://github.com/dotnet/roslyn/blob/2f6471ddabf64c00ecb7cdbf3a51e9cdc1d63f86/src/VisualStudio/CSharp/Impl/VSPackage.resx), it seems Visual Studio prefers "semicolon" over "semi-colon".

Reply:
Yes, the future enhancements to this feature is a good point. "Complete statement on semicolon" or "Complete statement with semicolon" sound fine to me ¯\\\_(ツ)_/¯

Reply:
> but obviously suffers in that its not quite everything this does, since I think it can also insert braces, right?

I don't think this command insert braces, but @sharwell to confirm.

Reply:
I like "Complete statement on semicolon", or maybe "Complete statement on typing semicolon" might  be even clearer?

Reply:
Oh, I like Sam's suggestion too of "Automatically complete statement on semicolon".